fenicsprecice/fenicsprecice.py

```diff
@@ -210,7 +214,7 @@ def read_data(self):
                 read_data = self._interface.read_block_vector_data(read_data_id, self._precice_vertex_ids)
 
             read_data = {tuple(key): value for key, value in zip(self._owned_vertices.get_coordinates(), read_data)}
-            read_data = communicate_shared_vertices(self._comm, self._rank, self._fenics_vertices, self._send_map,
+            read_data = communicate_shared_vertices(self._comm, self._comm.Get_rank(), self._fenics_vertices, self._send_map,
```

Comment:
I would suggest to change this function to: `communicate_shared_vertices(self._comm, self._fenics_vertices, self._send_map, self._recv_map, read_data)`

You can call `rank = comm.Get_rank()` inside of the function. Probably it's not possible to get rid of the `comm`?